tests/unit/AbstractTest.php

```diff
@@ -503,86 +502,9 @@ public function assertProduct(Dot $original_product, WC_Product $wc_product)
-            );
-        }
-
-        // Stock
-        $expected_in_stock = $original_product->get('flat_product.product.product_stock.in_stock');
-        if ($expected_in_stock) {
-            // Manage stock is based on stock unlimited. unlimited equals no stock management
-            $expected_manage_stock = !$original_product->get('flat_product.product.product_stock.unlimited');
-            $this->assertEquals(
-                $expected_manage_stock,
-                $wc_product->get_manage_stock(self::WC_CONTEXT_EDIT),
-                "[sku=$sku] WooCommerce manage stock doesn't match the expected manage stock"
-            );
-
-            // Stock quantity is set to one when stock isn't managed
-            $expected_stock_quantity = $expected_manage_stock ?
-                $original_product->get('flat_product.product.product_stock.value') : 1;
-            if (!$expected_manage_stock) {
-                // When we don't manage stock, the quantity will be null
-                $expected_stock_quantity = null;
-            }
-            $this->assertEquals(
-                $expected_stock_quantity,
-                $wc_product->get_stock_quantity(),
-                "[sku=$sku] WooCommerce stock quantity doesn't match the expected stock quantity"
-            );
-
-            // Stock status
-            $expected_stock_status = self::WC_STATUS_INSTOCK;
-            if (self::WC_TYPE_CONFIGURABLE === $wc_product->get_type()) {
-                $expected_stock_status = $expected_manage_stock ? self::WC_STATUS_INSTOCK : self::WC_STATUS_OUTOFSTOCK;
-            }
-
-            $this->assertEquals(
-                $expected_stock_status,
-                $wc_product->get_stock_status(),
-                "[sku=$sku] WooCommerce stock status doesn't match the expected stock status"
-            );
-        } else {
-            // Manage stock is always set to true when product is out of stock
-            $expected_manage_stock = true;
-            $this->assertEquals(
-                $expected_manage_stock,
-                $wc_product->get_manage_stock(self::WC_CONTEXT_EDIT),
-                "[sku=$sku] WooCommerce manage stock doesn't match the expected manage stock"
-            );
-
-            // Stock quantity is always set to 0 when product is out of stock
-            $expected_stock_quantity = 0;
-            $this->assertEquals(
-                $expected_stock_quantity,
-                $wc_product->get_stock_quantity(),
-                "[sku=$sku] WooCommerce stock quantity doesn't match the expected stock quantity"
-            );
-
-            // Stock status
-            $expected_stock_status = self::WC_STATUS_OUTOFSTOCK;
-            $this->assertEquals(
-                $expected_stock_status,
-                $wc_product->get_stock_status(),
-                "[sku=$sku] WooCommerce stock status doesn't match the expected stock status"
-            );
+        if ($isFullTest) {
```

Comment:
shouldn't you always test full product?  to make sure some data which should not be changes is still the same?

Reply:
Removed this as the initial PR is wrong.